rest/servers/spring/src/main/java/com/connexta/ingest/rest/spring/IngestApi.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.ingest.rest.spring;
+
+import com.connexta.ingest.rest.models.ErrorMessage;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import java.util.Optional;
+import javax.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RequestPart;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.multipart.MultipartFile;
+
+@Validated
+@Api(value = "Ingest", description = "the Ingest API")
+public interface IngestApi {
+  default Optional<NativeWebRequest> getRequest() {
+    return Optional.empty();
+  }
+
+  @ApiOperation(
+      value = "Ingest endpoint for products.",
+      nickname = "ingest",
+      notes =
+          "A system can use the Ingest endpoint to send one or more attachments and a JSON description of a product. ",
+      tags = {"ingest"})
+  @ApiResponses({
+    @ApiResponse(code = 202, message = "The ingest request was accepted. "),
+    @ApiResponse(
+        code = 400,
+        message =
+            "The client message could not understood by the server due to invalid format or syntax. ",
+        response = ErrorMessage.class),
+    @ApiResponse(
+        code = 401,
+        message = "The client could not be authenticated. ",
+        response = ErrorMessage.class),
+    @ApiResponse(
+        code = 403,
+        message = "The client does not have permission. ",
+        response = ErrorMessage.class),
+    @ApiResponse(
+        code = 501,
+        message = "The requested API version is not supported and therefore not implemented. ",
+        response = ErrorMessage.class),
+    @ApiResponse(
+        code = 200,
+        message = "Any other possible errors not currently known. ",
+        response = ErrorMessage.class)
+  })
+  @RequestMapping(
+      value = {"/ingest"},
+      produces = {"application/json"},
+      consumes = {"multipart/form-data"},
+      method = {RequestMethod.POST})
+  default ResponseEntity<Void> ingest(
```

Comment:
I think we should minimize changes between the generated code and what we commit in this file, so that it's easy to use the generator again once the bug is fixed. The generate code uses `ResponseEntity<Void>`.